packages/@aws-cdk/aws-efs/lib/efs-file-system.ts

```diff
@@ -0,0 +1,260 @@
+
+    /**
+     * The security groups/rules used to allow network connections to the file system.
+     */
+    public readonly connections: ec2.Connections;
+
+    /**
+     * @attribute
+     */
+    public readonly fileSystemID: string;
+
+    private readonly mountTargets: CfnMountTarget[] = [];
+    private readonly efsFileSystem: CfnFileSystem;
+
+    /**
+     * Constructor for creating a new EFS FileSystem.
+     */
+    constructor(scope: Construct, id: string, props: EfsFileSystemProps) {
+        super(scope, id);
+
+        if (props.throughputMode === EfsThroughputMode.PROVISIONED) {
+            if (props.provisionedThroughputInMibps === undefined) {
+                throw new Error('Property provisionedThroughputInMibps is required when throughputMode is PROVISIONED');
+            } else if (!Number.isInteger(props.provisionedThroughputInMibps)) {
+                throw new Error("Invalid input for provisionedThroughputInMibps");
+            } else if (props.provisionedThroughputInMibps < 1 || props.provisionedThroughputInMibps > 1024) {
+                this.node.addWarning("Valid values for throughput are 1-1024 MiB/s. You can get this limit increased by contacting AWS Support.");
+            }
+        }
+
+        this.efsFileSystem = new CfnFileSystem(this, "FileSystem", {
+            encrypted: props.encrypted,
+            kmsKeyId: (props.kmsKey ? props.kmsKey.keyId : undefined),
+            fileSystemTags: props.fileSystemTags,
+            lifecyclePolicies: (props.lifecyclePolicy ? Array.of({
+                transitionToIa: EfsLifecyclePolicyProperty[props.lifecyclePolicy]
+            } as CfnFileSystem.LifecyclePolicyProperty) : undefined),
+            performanceMode: props.performanceMode,
+            throughputMode: props.throughputMode,
+            provisionedThroughputInMibps: props.provisionedThroughputInMibps
+        });
+
+        this.fileSystemID = this.efsFileSystem.ref;
+        this.node.defaultChild = this.efsFileSystem;
+
+        const securityGroup = (props.securityGroup || new ec2.SecurityGroup(this, 'EfsSecurityGroup', {
+            vpc: props.vpc
+        }));
+
+        this.connections = new ec2.Connections({
+            securityGroups: [securityGroup],
+            defaultPort: ec2.Port.tcp(EfsFileSystem.DEFAULT_PORT)
+        });
+
+        const subnets = props.vpc.selectSubnets(props.vpcSubnets);
+
+        // We now have to create the mount target for each of the mentioned subnet
+        let mountTargetCount = 0;
+        subnets.subnetIds.forEach((subnetId: string) => {
+            const efsMountTarget = new CfnMountTarget(this,
+                "EfsMountTarget" + (++mountTargetCount),
+                {
+                    fileSystemId: this.fileSystemID,
+                    securityGroups: Array.of(securityGroup.securityGroupId),
+                    subnetId
+                });
+            this.mountTargets.push(efsMountTarget);
```

Comment:
Why are we holding on to these? Wouldn't we need their IDs out in some way?

Reply:
exposed both mount target's identifier and ip address.